pkg/plugins/layout.go

```diff
@@ -357,20 +357,11 @@ func (dev Disk) ReloadPartitionTable(console Console) error {
 }
 
 func (dev Disk) FindPartitionDevice(partNum int, console Console) (string, error) {
-	out, err := console.Run(fmt.Sprintf("lsblk -ltnpo name,type %s", dev))
+	out, err := console.Run(fmt.Sprintf("lsblk -ltnpo name %s%d", dev.Device, partNum))
```

Comment:
I am not sure this is always valid... you are assuming a device name pattern which I can't confirm it will be always correct.

In fact this is actually wrong for loop devices. Just did  a quick test:

```
$ sudo losetup -f --show partitioneddisk.raw
loop0
$ sudo partprobe /dev/loop0
$ ls /dev/loop0*
/dev/loop0  /dev/loop0p1  /dev/loop0p2  /dev/loop0p3  /dev/loop0p4
```
There is an additional `p`. That's why I prefer a tool that gives me the device name rather than trying to guess it.

Meanwhile I used `partprobe` because `blockdev` miserably failed in this example... 😞 could you verify it also? If blockdev also fails in your env I'd suggest to move back to partprobe and assume the ugly retry & wait logic is sufficient. This is all so frustrating...

Reply:
yiiikes

Reply:
yup, also fails for loop devices....

we do not create loop devices via yip rigth? in that case we could leave it in there anyways, but the change to cos-installer is wrong then.